x/concentrated-liquidity/spread_rewards.go

```diff
@@ -254,11 +255,40 @@ func (k Keeper) prepareClaimableSpreadRewards(ctx sdk.Context, positionId uint64
 	}
 
 	// Claim rewards, set the unclaimed rewards to zero, and update the position's accumulator value to reflect the current accumulator value.
-	spreadRewardsClaimed, forfeitedDust, err := updateAccumAndClaimRewards(spreadRewardAccumulator, positionKey, spreadRewardGrowthOutside)
+	spreadRewardsClaimedScaled, forfeitedDustScaled, err := updateAccumAndClaimRewards(spreadRewardAccumulator, positionKey, spreadRewardGrowthOutside)
 	if err != nil {
 		return nil, err
 	}
 
+	spreadFactorScalingFactor, err := k.getSpreadFactorScalingFactorForPool(ctx, position.PoolId)
+	if err != nil {
+		return nil, err
+	}
+
+	// We scale the spread factor per-unit of liquidity accumulator up to avoid truncation to zero.
+	// However, once we compute the total for the liquidity entitlement, we must scale it back down.
+	// We always truncate down in the pool's favor.
+	spreadRewardsClaimed := sdk.NewCoins()
+	forfeitedDust := sdk.DecCoins{}
+	if spreadFactorScalingFactor.Equal(sdk.OneDec()) {
+		// If the scaling factor is 1, we don't need to scale down the spread rewards.
+		// We also use the forfeited dust calculated updateAccumAndClaimRewards since it is already scaled down.
+		spreadRewardsClaimed = spreadRewardsClaimedScaled
+		forfeitedDust = forfeitedDustScaled
+	} else {
+		// If the scaling factor is not 1, we scale down the spread rewards and calculate the forfeited dust
+		// from the scaled down spread rewards (disregarding the forfeited dust calculated from updateAccumAndClaimRewards).
+		for _, coin := range spreadRewardsClaimedScaled {
+			scaledCoinAmt, truncatedAmt := scaleDownSpreadRewardAmount(coin.Amount, spreadFactorScalingFactor)
+			if !scaledCoinAmt.IsZero() {
+				spreadRewardsClaimed = append(spreadRewardsClaimed, sdk.NewCoin(coin.Denom, scaledCoinAmt))
+			}
+			if !truncatedAmt.IsZero() {
+				forfeitedDust = forfeitedDust.Add(sdk.NewDecCoinFromDec(coin.Denom, truncatedAmt))
+			}
+		}
+	}
```

Comment:
Another key area to focus on, specifically on how dust is handed here (incentives `prepareClaimAllIncentivesForPosition` never handled dust, whereas `prepareClaimableSpreadRewards` has always handled dust)